proto/ignition/msgs/statistic.proto

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "StatisticProtos";
+
+/// \ingroup ignition.msgs
+/// \interface Statistic
+/// \brief A message that contains statistics data.
+
+import "ignition/msgs/header.proto";
+
+message Statistic
+{
+  enum DataType
+  {
+    UNITIALIZED = 0;
+    AVERAGE      = 1;
+    MINIMUM      = 2;
+    MAXIMUM      = 3;
+    STDDEV       = 4;
+    SAMPLE_COUNT = 5;
+  }
+
+  /// \brief Optional header data
+  Header header  = 1;
+
+  DataType type  = 2;
+
+  string   name  = 3;
+
+  double   value = 4;
+}
+
+/// \brief A named group of statistics.
+message StatisticsGroup
+{
+  /// \brief Optional header data.
+  Header header                 = 1;
+
+  /// \brief Name of the group.
+  string name                   = 2;
+
+  /// \brief Statistics the belong to this group.
+  repeated Statistic statistics = 3;
+}
+
+message Metric
+{
+  /// \brief Optional header data
+  Header header  = 1;
+
+  /// \brief Unit of measurement such as seconds, meters, liters.
+  string   unit  = 2;
```

Comment:
Is the purpose of having `unit` here to group all values that share the same unit within the same metric? Otherwise I think we should move `unit` to `Statistic`.